File: src/test/java/test/ListRolesTest.java
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class ListRolesTest {

    private static final String ACCESS_KEY_ID = "YOUR_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "YOUR_ACCESS_KEY_SECRET";


    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions(ACCESS_KEY_ID, ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        ListRolesDto request = new ListRolesDto();
        request.setNamespace("namespace_6988");
        request.setPage(0);
        request.setLimit(0);

        RolePaginatedRespDto response = client.listRoles(request);
        System.out.println(JsonUtils.serialize(response));
    }

}

File: src/test/go/test/ListRolesTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.ListRolesDto{
        Keywords: "keywords_6550",
        Namespace: "namespace_3937",
        Page: 0,
        Limit: 0,
    }
    response := client.ListRoles(&request)
    fmt.Println(response)
}

